near/omni-bridge/src/lib.rs

```diff
@@ -517,11 +547,122 @@ impl Contract {
+                    token_id,
+                    storage_deposit_amount: fast_fin_transfer_msg.storage_deposit_amount,
+                };
+                PromiseOrValue::Promise(
+                    Self::check_or_pay_ft_storage(
+                        &deposit_action,
+                        &mut NearToken::from_yoctonear(storage_deposit_amount),
+                    )
+                    .then(
+                        Self::ext(env::current_account_id())
+                            .with_static_gas(VERIFY_PROOF_CALLBACK_GAS)
+                            .fast_fin_transfer_to_near_callback(fast_transfer, sender_id),
+                    ),
+                )
+            }
+            _ => {
+                self.fast_fin_transfer_to_other_chain(fast_transfer, sender_id);
+                PromiseOrValue::Value(U128(0))
+            }
+        }
+    }
+
+    #[private]
+    pub fn fast_fin_transfer_to_near_callback(
+        &mut self,
+        #[serializer(borsh)] fast_transfer: FastTransfer,
+        #[serializer(borsh)] relayer_id: AccountId,
+    ) -> Promise {
+        require!(
+            Self::check_storage_balance_result(1),
+            "STORAGE_ERR: The transfer recipient is omitted"
+        );
+
+        let OmniAddress::Near(recipient) = fast_transfer.recipient.clone() else {
+            env::panic_str("ERR_INVALID_STATE")
+        };
+
+        let required_balance = self
+            .add_fast_transfer(&fast_transfer, &relayer_id)
+            .saturating_add(ONE_YOCTO);
+        self.update_storage_balance(relayer_id, required_balance, NearToken::from_yoctonear(0));
+
+        self.send_tokens(
+            fast_transfer.token_id,
+            recipient,
+            U128(fast_transfer.amount.0 - fast_transfer.fee.fee.0),
+            fast_transfer.msg,
+        )
+    }
+
+    fn fast_fin_transfer_to_other_chain(
+        &mut self,
+        fast_transfer: FastTransfer,
+        relayer_id: AccountId,
+    ) {
+        let mut required_balance = self.add_fast_transfer(&fast_transfer, &relayer_id);
+
+        let destination_nonce =
+            self.get_next_destination_nonce(fast_transfer.recipient.get_chain());
+        self.current_origin_nonce += 1;
+
+        let transfer_message = TransferMessage {
+            origin_nonce: self.current_origin_nonce,
+            token: OmniAddress::Near(fast_transfer.token_id.clone()),
+            amount: fast_transfer.amount,
+            recipient: fast_transfer.recipient,
+            fee: fast_transfer.fee,
+            sender: OmniAddress::Near(relayer_id.clone()),
+            msg: fast_transfer.msg,
+            destination_nonce,
+            parent_transfer_id: Some(fast_transfer.transfer_id),
+        };
+
+        required_balance = self
+            .add_transfer_message(transfer_message, relayer_id.clone())
+            .saturating_add(required_balance);
+
+        self.update_storage_balance(relayer_id, required_balance, env::attached_deposit());
```

Comment:
`env::attached_deposit()` -- this value isn’t always zero, is it?

Reply:
Changed to 0